pcieutil/main.py

```diff
@@ -0,0 +1,199 @@
+def load_platform_pcieutil():
+    global platform_pcieutil
+    global hwsku_plugins_path
+    # Get platform and hwsku
+    (platform, hwsku) = get_platform_and_hwsku()
+
+    # Load platform module from source
+    try:
+        hwsku_plugins_path = "/".join([PLATFORM_ROOT_PATH, platform, "plugins"])
+        sys.path.append(os.path.abspath(hwsku_plugins_path))
+        from pcieutil import PcieUtil
+    except ImportError as e:
+        log_warning("Fail to load specific PcieUtil moudle. Falling down to the common implementation")
+        try:
+            from sonic_platform_base.sonic_pcie.pcie_common import PcieUtil
+            platform_pcieutil = PcieUtil(hwsku_plugins_path)
+        except ImportError as e:
+            log_error("Fail to load default PcieUtil moudle. Error :{}".format(str(e)), True)
+            raise e
+
+
+# ==================== CLI commands and groups ====================
+
+
+# This is our main entrypoint - the main 'psuutil' command
+@click.group()
+def cli():
+    """pcieutil - Command line utility for checking pci device"""
+    if os.geteuid() != 0:
+        click.echo("Root privileges are required for this operation")
+        sys.exit(1)
+
+    # Load platform-specific psuutil class
+    load_platform_pcieutil()
+
+# 'version' subcommand
+@cli.command()
+def version():
+    """Display version info"""
+    click.echo("pcieutil version {0}".format(VERSION))
+
+#show the platform PCIE info
+
+
+def print_test_title(testname):
+    click.echo("{name:=^80s}".format(name=testname))
+
+#  Show PCIE lnkSpeed
+@cli.command()
+def pcie_show():
+    '''Display PCIe Device '''
+    testname = "Display PCIe Device"
+    print_test_title(testname)
+    resultInfo = platform_pcieutil.get_pcie_device()
+    for item in resultInfo:
+        Bus = item["bus"]
+        Dev = item["dev"]
+        Fn = item["fn"]
+        Name = item["name"]
+        Id = item["id"]
+        print "bus:dev.fn %s:%s.%s - dev_id=0x%s,  %s" % (Bus,Dev,Fn,Id,Name) 
+        
+    
+
+
+
+#  Show PCIE Vender ID and Device ID
+@cli.command()
+def pcie_check():
+    '''Check PCIe Device '''
+    testname = "PCIe Device Check"
+    err = 0
+    print_test_title(testname)
+    resultInfo = platform_pcieutil.get_pcie_check()
+    for item in resultInfo:
+        if item["result"] == "Passed":
+            log_out(item["name"], "Passed")
+        else:
+            log_out(item["name"], "Failed")
+            log_warning("PCIe Device: " +  item["name"] + " Not Found")
+            err+=1
+    if err:
+        print "PCIe Device Checking All Test ----------->>> FAILED"
+    else:
+        print "PCIe Device Checking All Test ----------->>> PASSED"
+        
+
+
+
+@cli.command()
+@click.confirmation_option(prompt="Are you sure to overwrite config file pcie.yaml with current pcie device info?")
+def pcie_generate():
+    '''Generate config file with current pci device'''
+    platform_pcieutil.dump_conf_yaml()
```

Comment:
aren't PCIe devices fixed for platform? What is the baseline to generate the configuration file? Shouldn't we request the platform vendors to provide their expected pcie.yaml files for each platform types?
And do you have any reference or example of the configuration file - pcie.yaml?